x-pack/legacy/plugins/infra/public/components/logging/log_entry_flyout/log_entry_actions_menu.tsx

```diff
@@ -91,3 +105,33 @@ const getUptimeLink = (logItem: InfraLogItem) => {
     hash: `/?search=(${searchExpressions.join(' OR ')})`,
   });
 };
+
+const getAPMLink = (logItem: InfraLogItem) => {
+  const traceIdEntry = logItem.fields.find(
+    ({ field, value }) => value != null && field === 'trace.id'
+  );
+
+  if (!traceIdEntry) {
+    return undefined;
+  }
+
+  const nowInSeconds = Math.floor(Date.now() / 1000);
+  const timestampField = logItem.fields.find(({ field }) => field === '@timestamp');
+  const timestampInSeconds = timestampField
+    ? Math.floor(Date.parse(timestampField.value) / 1000)
+    : null;
+  const { rangeFrom, rangeTo } = timestampInSeconds
+    ? (() => {
+        const timeAgoPlus1Minute = nowInSeconds - timestampInSeconds + 60;
+        const timeAgo = nowInSeconds - timestampInSeconds;
+        return { rangeFrom: `now-${timeAgoPlus1Minute}s`, rangeTo: `now-${timeAgo}s` };
```

Comment:
Nice, a few things here:

* Right now `rangeFrom` is 60s ahead of `rangeTo` which would need to be switched.
* Because a log can happen at any time during a trace and a trace can run for a while, we should be a bit more lenient in our time range. Maybe 10 min on either side of the log timestamp?
* The seconds-based time range is nice (no moment dependency) but a bit hard to read if someone hovers over the time picker once they get to APM ("-765s?"). 

What about something like:
```js
const rangeFrom = new Date();
const rangeTo = new Date(rangeFrom);

rangeFrom.setMinutes(rangeFrom.getMinutes() - 10)
rangeTo.setMinutes(rangeTo.getMinutes() + 10)

{ rangeFrom: rangeFrom.toISOString(), rangeTo: rangeTo.toISOString() }
```

Reply:
> * Right now `rangeFrom` is 60s ahead of `rangeTo` which would need to be switched.

Oops no you have it right as-is, n/m. Confusing with the subtraction. :)